libdevcore/SQLConnectionPool.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2019 fisco-dev contributors.
+ */
+/** @file SQLConnectionPool.cpp
+ *  @author darrenyin
+ *  @date 2019-04-25
+ */
+
+#include "SQLConnectionPool.h"
+
+using namespace dev::db;
+using namespace std;
+
+int SQLConnectionPool::InitConnectionPool(const std::string& strDbType, const std::string& strDbIp,
+    uint32_t dwDbPort, const std::string& strDbUser, const std::string& strDbPasswd,
+    const std::string& strDbname, const std::string& strDbCharset, uint32_t dwInitConnections,
+    uint32_t dwMaxConnection)
+{
+    if (strDbType == "mysql")
+    {
+        char cConnectionBuf[2048] = {0};
+        snprintf(cConnectionBuf, sizeof(cConnectionBuf),
+            "mysql://%s:%u/%s?user=%s&password=%s&charset=%s", strDbIp.c_str(), dwDbPort,
+            strDbname.c_str(), strDbUser.c_str(), strDbPasswd.c_str(), strDbCharset.c_str());
+        string strConnetionBuf = cConnectionBuf;

```

Comment:
why does it need to transfer from char* to string and reverse again ?

Reply:
changed.